tests/models/test_implicit_knn.py

```diff
@@ -154,10 +154,30 @@ def test_with_whitelist(self, dataset: Dataset, filter_viewed: bool, expected: p
             ),
         ),
     )
-    def test_i2i(
-        self, dataset: Dataset, filter_itself: bool, whitelist: tp.Optional[np.ndarray], expected: pd.DataFrame
-    ) -> None:
+    def test_i2i(self, filter_itself: bool, whitelist: tp.Optional[np.ndarray], expected: pd.DataFrame) -> None:
         base_model = TFIDFRecommender(K=5, num_threads=2)
+        # Recreate dataset to prevent same co-occurrence count between (11, 14) and (11, 15)
+        # which leads to different results in the test in Python 3.13
```

Comment:
You mean for the default dataset the results in python 3.8-3.12 and 3.13 are different?  
Do you have any idea why is this happening?  
Probably the problem is not with the python itself, but with some of the dependencies?

Reply:
Yes, the behavior of implicit ItemKNN changes due to the changes in the similarity matrix.

Considering the cosign similarity, co-occurrence between 11-14 and 11-15 are both 2 times, thus it causes flakiness.

Reply:
Let us take some time to examine this case carefully